.github/workflows/codeql_analysis.yml

```diff
@@ -0,0 +1,36 @@
+on:
+  workflow_dispatch:
+  schedule:
+    #        ┌───────────── minute (0 - 59)
+    #        │  ┌───────────── hour (0 - 23)
+    #        │  │ ┌───────────── day of the month (1 - 31)
+    #        │  │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+    #        │  │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        *  * * * *
+    - cron: '30 1 * * *'
```

Comment:
I'm not sure we should have it periodically. Upon merge and on PRs should be sufficient.

Reply:
I think a cron schedule is actually very useful. Often you will not introduce a security issue on merge, but it will pop up without code changes, simply because it is later discovered that it was already there but not known yet.

This is especially of interest if we don't merge anything for a while.

Reply:
What should be the next step here folks?

Reply:
@jpkrohling @Oberon00

Reply:
@Oberon00 has a good point, feel free to keep the scheduled runs.

Reply:
@sakshi-1505 / @jpkrohling is this still relevant?